### ISSUE TITLE
const noexcept override try

### ISSUE THREAD

**Issue by hennna** [opentracing/opentracing-cpp#60]
I'm getting compilation error "expected ';' at end of declaration list" from the following line of code: https://github.com/opentracing/opentracing-cpp/blob/master/include/opentracing/tracer.h#L219

I'm not familiar with the expected behavior of this code. Is wrapping with curly braces an option?

```
void Apply(StartSpanOptions& options) const noexcept override {
 try {
    if (referenced_) options.references.emplace_back(type_, referenced_);
  } catch (const std::bad_alloc&) {
    // Ignore reference if memory can't be allocated for it.
  }
}
```

**Comment by rnburn** [comment]
Curious, which compiler gives that error? It should be valid C++, but your snippet behaves the same so I'm fine changing to that.

**Comment by hennna** [comment]
I'm using google's internal version of bazel. I believe it uses clang.

**Comment by rnburn** [comment]
Thanks.

There's CI coverage for [clang](https://github.com/opentracing/opentracing-cpp/blob/master/.travis.yml#L9), but I guess google must be using a different branch.

**Comment by rnburn** [comment]
Fixed in #61
